core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/ResponseTransformer.java

```diff
@@ -129,6 +129,8 @@ default boolean needsConnectionLeftOpen() {
 
                 // Retry the request
                 throw RetryableException.builder().message(copyError).cause(copyException).build();
+            } finally {
+                IoUtils.closeQuietly(in, null);
```

Comment:
Shouldn't the SDK be closing the stream, not the ResponseTransformer implementation?

Reply:
We should be closing it here if the needsConnectionLeftOpen returns false.

https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/HandleResponseStage.java#L134

Reply:
Ahhh, right! nvm then.